app/src/main/java/io/apicurio/registry/rules/compatibility/CompatibilityExecutionResult.java

```diff
@@ -0,0 +1,40 @@
+package io.apicurio.registry.rules.compatibility;
+
+import io.apicurio.registry.rules.compatibility.jsonschema.diff.Difference;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Created by aohana
+ */
+public class CompatibilityExecutionResult {
+
+    private boolean isCompatible;
+    private Set<Difference> incompatibleDifferences;
+
+    public CompatibilityExecutionResult(boolean isCompatible, Set<Difference> incompatibleDifferences) {
+        this.isCompatible = isCompatible;
+        this.incompatibleDifferences = incompatibleDifferences;
+    }
+
+    public boolean isCompatible() {
+        return isCompatible;
+    }
+
+    public void setCompatible(boolean compatible) {
+        isCompatible = compatible;
+    }
+
+    public Set<Difference> getIncompatibleDifferences() {
+        return incompatibleDifferences;
+    }
+
+    public void setIncompatibleDifferences(Set<Difference> incompatibleDifferences) {
+        this.incompatibleDifferences = incompatibleDifferences;
+    }
+
+    public static CompatibilityExecutionResult empty(boolean isCompatible) {
```

Comment:
When should someone return an empty compatibility result? What does it mean?

Reply:
all the schema types (except JSON) currently don't have information on the compatibility error so they will return isCompatible (true/false) and an empty diff set